server/embedded/src/org/labkey/embedded/LabKeyServer.java

```diff
@@ -0,0 +1,358 @@
+								}
+							}
+
+							if (!foundDistributionZip)
+							{
+								throw new ConfigException("Unable to find distribution zip required to run LabKey server.");
+							}
+						}
+						catch (IOException | ConfigException e)
+						{
+							throw new RuntimeException(e);
+						}
+					}
+				}
+				else
+				{
+					webAppLocation = props.getWebAppLocation();
+				}
+
+				// TODO : 8021 :fix context path - put app at root by default
+				StandardContext context = (StandardContext)tomcat.addWebapp("/labkey", webAppLocation);
+
+                // Push the JDBC connection for the primary DB into the context so that the LabKey webapp finds them
+				getDataSourceResources(props).forEach(contextResource -> context.getNamingResources().addResource(contextResource));
+				// Add the SMTP config
+				context.getNamingResources().addResource(getMailResource());
+
+				// And the master encryption key
+				context.addParameter("MasterEncryptionKey", props.getMasterEncryptionKey());
+
+				// Point at the special classloader with the hack for SLF4J
+				WebappLoader loader = new WebappLoader();
+				loader.setLoaderClass(LabKeySpringBootClassLoader.class.getName());
+				context.setLoader(loader);
+				context.setParentClassLoader(this.getClass().getClassLoader());
+
+				return super.getTomcatWebServer(tomcat);
+			}
+
+			private List<ContextResource> getDataSourceResources(ContextProperties props)
+			{
+				List<ContextResource> dataSourceResources = new ArrayList<>();
+				var numOfDataResources = props.getUrl().size();
+
+				for (int i = 0; i < numOfDataResources; i++)
+				{
+					ContextResource dataSourceResource = new ContextResource();
+					dataSourceResource.setName(props.getDataSourceName().get(i));
+					dataSourceResource.setAuth("Container");
+					dataSourceResource.setType(DataSource.class.getName());
+					dataSourceResource.setProperty("driverClassName", props.getDriverClassName().get(i));
+					dataSourceResource.setProperty("url", props.getUrl().get(i));
+					dataSourceResource.setProperty("password", props.getPassword().get(i));
+					dataSourceResource.setProperty("username", props.getUsername().get(i));
+					dataSourceResource.setProperty("maxTotal", "20");
+					dataSourceResource.setProperty("maxIdle", "10");
+					dataSourceResource.setProperty("maxWaitMillis", "120000");
+					dataSourceResource.setProperty("accessToUnderlyingConnectionAllowed", "true");
+					dataSourceResource.setProperty("validationQuery", "SELECT 1");
+					dataSourceResources.add(dataSourceResource);
+				}
+
+				return  dataSourceResources;
+			}
+
+			private ContextResource getMailResource()
+			{
+				// Get session/mail properties
+				MailProperties mailProps = smtpSource();
+				ContextResource mailResource = new ContextResource();
+				mailResource.setName("mail/Session");
+				mailResource.setAuth("Container");
+				mailResource.setType("javax.mail.Session");
+				mailResource.setProperty("mail.smtp.host", mailProps.smtpHost);
```

Comment:
We'll need more properties here too. At least the ones captured here:

https://www.labkey.org/Documentation/wiki-page.view?name=cpasxml#SMTPsettings